### DISCUSSION TITLE
Using theme.colors directly won't affect the output

### DISCUSSION THREAD

**Discussion by andree-gojek** [tailwindlabs/tailwindcss#12385] [Help]
**What version of Tailwind CSS are you using?**

v3.3.5

**What build tool (or framework if it abstracts the build tool) are you using?**

Vite v4.2.1

**What version of Node.js are you using?**

v18.14.12

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

**Describe your issue**

The colors are not overridden. If I'm using `theme.extend.colors`, it will work. But if I use `theme.colors` directly, it won't be applied. The tailwind default colors will take precedence.

<img width="783" alt="Screenshot 2023-11-09 at 11 06 54" src="https://github.com/tailwindlabs/tailwindcss/assets/135572821/a3e6c10c-c532-441c-abaa-17f372dc269c">

I am using tailwind.config.ts. The output of the intellisense not showing any errors. I wonder why is this happening?

<img width="1360" alt="Screenshot 2023-11-09 at 11 16 52" src="https://github.com/tailwindlabs/tailwindcss/assets/135572821/3155e72b-82e9-47f9-8fe1-adfed26b6201">

**Comment by alex-krasikau** [comment]
Hey @andree-gojek!

Could you please provide an example for reproduction?

I tried it myself, and the case you described seems to work fine for me - https://play.tailwindcss.com/wEi4xXe9GL?file=config.
